### PR TITLE
ddtrace/tracer: Switch default context propagation order

### PR DESCRIPTION
**Pull request by JianyiGao**
### What does this PR do?
This PR switch default context propagation order from tracecontext,datadog to datadog,tracecontext
[The same change is merged into `v2-dev` branch](https://github.com/DataDog/dd-trace-go/pull/2349), this PR is to cherry pick the same commit to `main` so system test can run with the new default context propagation order.
System test change: https://github.com/DataDog/system-tests/pull/1853

### Reviewer's Checklist

- [x] Changed code has unit tests for its functionality at or near 100% coverage.
- [x] There is a benchmark for any new code, or changes to existing code.
- [x] If this interacts with the agent in a new way, a system test has been added.
- [x] Add an appropriate team label so this PR gets put in the right place for the release notes.

For Datadog employees:

- [x] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

**Comment by JianyiGao** [comment]
@dianashevchenko suggested to add this change to main, we can also test the new default behavior on system tests. I am not aware that this is a breaking change, would you like to chime in Diana?

**Comment by felixge** [comment]
> @JianyiGao Just curious: why is this change going to be merged into main? It's a breaking change.

AFAIK this is not a breaking change. See PROF-8549 for my notes on this or let's discuss in slack.

**Comment by mabdinur** [comment]
Parametric tests pass with the following commit: https://github.com/DataDog/dd-trace-go/pull/2368/commits/288c76de071b6ab39a9b31b1ca788715238f60ed

**Review by katiehockman — APPROVED** [review]

File: .github/workflows/parametric-tests.yml [hunk 1]
```diff
@@ -33,7 +33,8 @@ jobs:
         uses: actions/checkout@v3
         with:
           repository: 'DataDog/system-tests'
-          ref: ${{ inputs.ref }}
+          # https://github.com/DataDog/system-tests/pull/1853
+          ref: 'ff2b0873b1414a199f5db75976c8aa06ff918aff'
```
Comment (mabdinur) [inline .github/workflows/parametric-tests.yml:37]:
```suggestion
          ref: ${{ inputs.ref }}
```

**Review by mabdinur — APPROVED** [review]

**Comment by mabdinur** [comment]
@katiehockman when you get a chance can you merge this PR 🥳

**Pull request merged**